cli/packages/cmd/user.go

```diff
@@ -0,0 +1,248 @@
+	Run:                   func(cmd *cobra.Command, args []string) {},
+}
+
+var switchCmd = &cobra.Command{
+	Use:                   "switch",
+	Short:                 "Used to switch between Infisical profiles",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical switch",
+	Args:                  cobra.ExactArgs(0),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		util.RequireLogin()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		//get previous logged in profiles
+		loggedInProfiles, err := getLoggedInUsers()
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Unable to get logged Profiles")
+		}
+
+		//prompt user
+		profile, err := LoggedInUsersPrompt(loggedInProfiles)
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Prompt error")
+		}
+
+		//write to config file
+		configFile, err := util.GetConfigFile()
+		if err != nil {
+			util.HandleError(err, "[infisical user switch]: Unable to get config file")
+		}
+
+		configFile.LoggedInUserEmail = profile
+
+		//set logged in user domain
+		ok := util.ConfigContainsEmail(configFile.LoggedInUsers, profile)
+
+		if !ok {
+			//profile not in loggedInUsers
+			configFile.LoggedInUsers = append(configFile.LoggedInUsers, models.LoggedInUser{
+				Email:  profile,
+				Domain: config.INFISICAL_URL,
+			})
+			//set logged in user domain
+			configFile.LoggedInUserDomain = config.INFISICAL_URL
+
+		} else {
+			//exists, set logged in user domain
+			for _, v := range configFile.LoggedInUsers {
+				if profile == v.Email {
+					configFile.LoggedInUserDomain = v.Domain
+					break
+				}
+			}
+		}
+
+		err = util.WriteConfigFile(&configFile)
+		if err != nil {
+			util.HandleError(err, "")
+		}
+	},
+}
+
+var updateCmd = &cobra.Command{
+	Use:                   "update",
+	Short:                 "Used to update properties of an Infisical profile",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical user update",
+	Args:                  cobra.ExactArgs(0),
+	Run:                   func(cmd *cobra.Command, args []string) {},
+}
+
+var domainCmd = &cobra.Command{
+	Use:                   "domain",
+	Short:                 "Used to update the domain of an Infisical profile",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical user update domain",
+	Args:                  cobra.ExactArgs(0),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		util.RequireLogin()
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		//prompt for profiles selection
+		loggedInProfiles, err := getLoggedInUsers()
+		if err != nil {
+			util.HandleError(err, "[infisical user update domain]: Unable to get logged Profiles")
+		}
+
+		//prompt user
+		profile, err := LoggedInUsersPrompt(loggedInProfiles)
+		if err != nil {
+			util.HandleError(err, "[infisical user update domain]: Prompt error")
+		}
+
+		domain := ""
+		domainQuery := true
+		if config.INFISICAL_URL_MANUAL_OVERRIDE != util.INFISICAL_DEFAULT_API_URL {
```

Comment:
Can you explain why you choose to use `INFISICAL_URL_MANUAL_OVERRIDE` variable instead of   `config.INFISICAL_URL`? I see in `GetCurrentLoggedInUserDetails`, you set `INFISICAL_URL_MANUAL_OVERRIDE=config.INFISICAL_URL` so i'm curious if we need this or if there is a use case I didn't catch

Reply:
Right, so with the two manual override methods, `config.INFISICAL_URL` gets updated with the user self-hosted-instance url from either `--domain` or `INFISCAL_API_URL` env var  before it hits the run function of any infisical command, now if you notice in the `PreRun` or `Run` of most commands there is a check for a user's logged in status in `GetCurrentLoggedInUserDetails`. For multi-profiles the config file holds the current logged in user as well as the user's domain. Now in here: https://github.com/Infisical/infisical/pull/481/commits/d4206cdbd82707d5ed3e655bf5e1012c1241d619#diff-eb3a7252b09a5793a44fec4d5e033dda44f2fc86484c3d1a728bb57208058fdaR93 , there is a check for if the config data structure holds a `LoggedInUserDomain` value. If there is it overrides the `config.INFISICAL_URL`. To still support the two manual override methods we don't want that. So the `config.INFISICAL_URL_MANUAL_OVERRIDE` gets assigned the `config.INFISICAL_URL` value before its updated. This is necessary because we can use that to check if there was a domain override from those two methods in the `Run` functions of the relevant commands, in this case `login` and `user update domain`.

Reply:
Then we check if there was an override by comparing `config.INFISICAL_URL_MANUAL_OVERRIDE` with the default api

Reply:
Oh i see, makes sense. So INFISICAL_URL_MANUAL_OVERRIDE holds the value defined by the two methods and is used to detect if those two methods are used. Thanks for explaining